mlrun/api/api/utils.py

```diff
@@ -438,10 +454,13 @@ def ensure_function_has_auth_set(function, auth_info: mlrun.api.schemas.AuthInfo
                 ]
 
             function.metadata.credentials.access_key = auth_info.access_key
-        if not function.metadata.credentials.access_key:
+
+        if raise_on_access_key_not_set and not function.metadata.credentials.access_key:
```

Comment:
If `function.metadata.credentials.access_key = None` and `raise_on_access_key_not_set = False` you will get to the next `if` that tests `if not function.metadata.credentials.access_key.startswith(...` that will explode because `None` has no `startswith`.
You should keep the previous `if` and check the `raise_on_access_key_not_set` inside - explode if `True`, else `return`.